markdown.tmLanguage.base.yaml

```diff
@@ -108,9 +108,9 @@ repository:
         - {include: text.html.derivative}
         while: (?i)^(?!.*</(script|style|pre)>)
     - begin: (?i)(^|\G)\s*(?=</?[a-zA-Z]+[^\s/&gt;]*(\s|$|/?>))
+      end: ^(?=\s*$)
       patterns:
       - {include: text.html.derivative}
-      while: ^(?!\s*$)
```

Comment:
`while` is used here to prevent the html grammar from escaping outside of the html block in the markdown file. For example, for the markdown:

```

<div a=">
    bla
</div>

Rest of the file
```

The entire rest of the html is parsed as an attribute because there is no closing `"`:

<img width="330" alt="Screen Shot 2020-04-08 at 6 02 15 PM" src="https://user-images.githubusercontent.com/12821956/78847250-19673800-79c3-11ea-8886-22ae886866a8.png">


However the `while` does stop it so that the html grammar doesn't escape. With `end` however, the html grammar continues parsing the rest of the file until it finds a `"`:

<img width="291" alt="Screen Shot 2020-04-08 at 6 03 02 PM" src="https://user-images.githubusercontent.com/12821956/78847315-4ae00380-79c3-11ea-99aa-7a5cf82856c7.png">

So I don't feel that changing end -> while is the proper fix. Do you know why `while` doesn't work?